### ISSUE TITLE
$httpBackend cannot mock binary blob responses at IE11 (InvalidStateError)

### ISSUE THREAD

**Issue by yneves** [angular/angular.js#9669]
I found this issue running tests with karma at IE11 / Win8.1.

When i try to mock a blob response with `$httpBackend.expect`, i got `InvalidStateError`.

I dig and found that this happens because of the `angular.copy` which clones fake response data.

If i hack `angular.copy` with something like this the error stops.

```js
if (isBlob(source)) {
    destination = new Blob([source]);
} else if (isArray(source)) {
```

**Comment by jurko-gospodnetic** [comment]
This is not browser specific. And pretty much prevents your from mocking HTTP requests with `Blob` responses. :-(

**Comment by gkalpak** [comment]
@yneves, I couldn't reprocude it just by `angular.copy`ing and `Blob`. Could you create a live demo of the issue (e.g. using CodePen, Plnkr etc) ?

**Comment by jurko-gospodnetic** [comment]
Here's a plnkr reproducing the error. Shows that you can't treat the almost-blob API response object you get from ng-mock as a true blob, while you can do so with original blob given to ng-mock in its expectation.

http://plnkr.co/xenwdKodooTrYO7VG2P2

**Comment by gkalpak** [comment]
@jurko-gospodnetic, thx for the reproduction. I have submitted a fix in #14064.
